xarray/core/missing.py

```diff
@@ -734,6 +734,13 @@ def interp_func(var, x, new_x, method, kwargs):
         # if usefull, re-use localize for each chunk of new_x
         localize = (method in ["linear", "nearest"]) and (new_x[0].chunks is not None)
 
+        # scipy.interpolate.interp1d always forces to float.
```

Comment:
don't we also support the numpy interpolator? Should we return dtype from `_get_interpolator` to account for both cases?

Reply:
Maybe, but since vectorizable_only is set to True always we cant get to the numpyinterpolator, atleast in interp_func. Does np.interp also force float?

Knowing the dtype is only relevant for dask arrays also which would mean unneccesary rows of code for non-dasks.

I also have some dreams of supporting the correct dtypes in dataset.interp which would probably mean overwriting that suggested dtype anyway.

Reply:
> Does np.interp also force float?

Yes, according to the docstring, so my comment is not relevant. Thanks.